SPARQL.md

```diff
@@ -0,0 +1,84 @@
+## SPARQL Queries and Parameterization
+
+As the Pan-Archival Catalogue Services (PACS) rely on an RDF store to hold the catalogue data, the W3C recommended
+[SPARQL Query Language](https://www.w3.org/TR/sparql11-query/) is the standardised way to query this data.
+
+The SPARQL queries used by PACS are kept in the `src/main/resources/sparql` directory of the project.
+
+An architectural decision was made to keep the SPARQL in separate files with the standard `.rq` SPARQL file extension,
+rather than embed the SPARQL into the Scala source code that makes up PACS. This allows the queries to be developed,
+tested and debugged in isolation from the rest of the application. That being said, there is still a requirement to
+inject values into these queries from within the application so that they can be re-used to handle query variations.
+
+Within PACS we use the [Apache Jena ARQ API](https://jena.apache.org/documentation/query/index.html) to manage and
+manipulate the RDF data within the data store. This library includes a [ParameterizedSparqlString](https://jena.apache.org/documentation/javadoc/arq/org.apache.jena.arq/org/apache/jena/query/ParameterizedSparqlString.html)
+class which forms the basis of our approach to query parameterization. One nice feature of `ParameterisedSparqlString`
+is that it can be initialised with an existing query and then used to replace variables within the query with
+specific values. For example, the following query, which will select all RDF triples in the RDF store, can be 
+parameterized by replacing `?predicateParam` with `rdf:type`:
+```
```

Comment:
For your code examples can you add the language annotation so that GitHub colourizes them, e.g.:

```sparql
SELECT ?subject ?object
WHERE { ?subject ?predicateParam ?object . }
```

as opposed to:
```
SELECT ?subject ?object
WHERE { ?subject ?predicateParam ?object . }
```

To add the syntax allocation, just follow the 3 back-ticks with the syntax name e.g. `sparql`:
```
```sparql
```

Reply:
Thanks, didn't realise it recognised SPARQL!